// react-vitals/dist/cjs/Slice.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Slice = void 0;
const react_1 = __importDefault(require("react"));
function Slice({ width, height, size = 50, children, className, style }) {
    const column = Math.ceil(width / size);
    const row = Math.ceil(height / size);
    const cells = [];
    for (let i = 0; i < row; i++) {
        for (let j = 0; j < column; j++) {
            cells.push(react_1.default.createElement("div", { key: i + '-' + j, style: {
                    position: 'absolute',
                    width: size,
                    height: size,
                    overflow: 'hidden',
                    left: j * size,
                    top: i * size,
                } },
                react_1.default.createElement("div", { style: { position: 'absolute', left: -j * size, top: -i * size } }, children)));
        }
    }
    return (react_1.default.createElement("div", { className: className, style: Object.assign(Object.assign({ position: 'relative', overflow: 'hidden' }, style), { width, height }) }, cells));
}
exports.Slice = Slice;
